analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotUseOutRefParameters.cs

```diff
@@ -25,29 +25,29 @@
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Extensions;
 using SonarAnalyzer.Helpers;
 
 namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     public sealed class DoNotUseOutRefParameters : SonarDiagnosticAnalyzer
     {
-        internal const string DiagnosticId = "S3874";
+        private const string DiagnosticId = "S3874";
         private const string MessageFormat = "Consider refactoring this method in order to remove the need for this '{0}' modifier.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DescriptorFactory.Create(DiagnosticId, MessageFormat);
+        private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
                     var parameter = (ParameterSyntax)c.Node;
 
-                    if (!parameter.Modifiers.Any(IsRefOrOut))
+                    if (!parameter.Modifiers.Any(IsRefOrOut)
+                        || IsDeconstructor((MethodDeclarationSyntax)parameter.Parent.Parent))
```

Comment:
`parameter.Parent.Parent` should always be a MethodDeclaration.

Reply:
Initially, I thought that this is not true in the case of records  `record Sample(int X)`  but the previous condition makes it safe I think.